lib/streamlit/elements/text_widgets.py

```diff
@@ -269,13 +278,19 @@ def _text_input(
             text_input_proto.set_value = True
 
         self.dg._enqueue("text_input", text_input_proto)
+        if value is None or (
+            check_type(widget_state.value) is str and widget_state.value == "None"
```

Comment:
Why is the `widget_state.value` `None` here? I would assume the value is just an empty string or?

Reply:
Sometimes it will be an empty `string`, but what if the value is `None`? It should follow this conditional then